include/maybe.hrl

```diff
@@ -10,7 +10,7 @@
 
 -type ok(A)       :: {ok, A}.
 -type error(A)    :: {error, A}.
--type maybe(A, B) :: ok(A) | error(B).
+-type 'maybe'(A, B) :: ok(A) | error(B).
```

Comment:
**Super unimportant tangent**: If this module and type had been given the correct name (`result`) from the start we would have to resort to these apostrophes. But there is of course no guarantee `result` would not be reserved in another universe, or `either` which would also have been a sense-making name for this type.

The original stdlib2 authors chose to name the [Result type](https://en.wikipedia.org/wiki/Result_type) "maybe" for some reason. `Maybe` is the Haskell name of the [Option type](https://en.wikipedia.org/wiki/Option_type). Haskell of course has a result type as well, but it is called `Either`. So this `s2_maybe` business is very confusing.

Anywho, I don't see us renaming it here "just because" it would fix the apostrophe inconvenience...

Reply:
Yeah, the naming is what it is, but making a name change that is non-backwards compatible seems worse than making a backwards compatible (but ugly) change